docs/hooks.py

```diff
@@ -0,0 +1,79 @@
+import os
+import re
+
+from markdown_refdocs.main import extract_to_markdown
+from mavis.annotate.constants import DEFAULTS as ANNOTATION_DEFAULTS
+from mavis.cluster.constants import DEFAULTS as CLUSTER_DEFAULTS
+from mavis.config import REFERENCE_DEFAULTS
+from mavis.illustrate.constants import DEFAULTS as ILLUSTRATION_DEFAULTS
+from mavis.pairing.constants import DEFAULTS as PAIRING_DEFAULTS
+from mavis.schedule.constants import OPTIONS as SUBMIT_OPTIONS
+from mavis.summary.constants import DEFAULTS as SUMMARY_DEFAULTS
+from mavis.util import ENV_VAR_PREFIX
+from mavis.validate.constants import DEFAULTS as VALIDATION_DEFAULTS
+
+
+def generate_settings_doc():
+    dirname = os.path.dirname(os.path.abspath(__file__))
+
+    for (filepath, title, namespaces) in [
+        (
+            'configuration/settings.md',
+            'Configurable Settings',
+            [
+                SUBMIT_OPTIONS,
+                REFERENCE_DEFAULTS,
+                SUMMARY_DEFAULTS,
+                PAIRING_DEFAULTS,
+                ANNOTATION_DEFAULTS,
+                VALIDATION_DEFAULTS,
+                CLUSTER_DEFAULTS,
+                ILLUSTRATION_DEFAULTS,
+            ],
+        ),
+    ]:
+        fname = os.path.join(dirname, filepath)
+        print('writing:', fname)
+        with open(fname, 'w') as fh:
+            fh.write(f'\n\n# {title}\n')
+            glossary = {}
+            for namespace in namespaces:
+                for term, value in namespace.items():
+                    typ = namespace.type(term).__name__
+                    # typ = CUSTOM_TYPES.get(typ, typ)
+                    desc = re.sub(r"\.?$", "", namespace.define(term, "")).capitalize()
+                    accepted = ''
+                    try:
+                        accepted = '\n\n**accepted values**: {}\n'.format(
+                            ', '.join(['`{}`'.format(repr(v)) for v in namespace.type(term).values()])
+                        )
+                    except AttributeError:
+                        pass
+                    defn = f'''## {term}
+
+**type**: `#!python {typ}`
+
+**environment variable**: `{ENV_VAR_PREFIX}{term.upper()}`
+
+**default**: `#!python {repr(value)}`{accepted}
+
+{desc}
+        '''
```

Comment:
Consider using `textwrap.dedent()` to keep the string inline with the rest of the code without adding any extra whitespace. I think it's a standard python library.

Reply:
cool! I didn't know there was a function for that! I have similar strings in other parts of the code tho so I will make an issue so we can replace them all at once instead of just here. created #236